librz/analysis/var_global.c

```diff
@@ -56,75 +77,132 @@ RZ_API RZ_OWN bool rz_analysis_var_global_add(RzAnalysis *analysis, RZ_NONNULL R
 	return true;
 }
 
+/**
+ * \brief Free the global variable instance except the flag
+ * 
+ * \param glob Global variable instance
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_free_except_flag(RZ_NONNULL RzAnalysisVarGlobal *glob) {
```

Comment:
Why is this necessary? I think the flag should never be freed, though it should always be unset, so it is ok to just keep one _free function. Any reason I'm missing?

Reply:
We need a function which frees the key-value pair of the global from the analysis hash table. This is that function. Of course, there might be another way to do this without introducing this function. I'll try to find one.